pkgs/development/python-modules/mkdocs/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, python3
+, fetchFromGitHub
+, buildPythonPackage
+, fetchpatch
+, click
+, jinja2
+, markdown
+, mergedeep
+, pyyaml
+, pyyaml-env-tag
+, ghp-import
+, importlib-metadata
+, watchdog
+, packaging
+, Babel
+, mock
+, pytestCheckHook
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  pname = "mkdocs";
+  version = "1.2.3";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-LBw2ftGyeNvARQ8xiYUho8BiQh+aIEqROP51gKvNxEo=";
+  };
+
+  patches = [
+    # TODO: remove this patch on the next release
+    # the upstream patch https://github.com/mkdocs/mkdocs/commit/c93fc91e4dc0ef33e2ea418aaa32b0584a8d354a.patch
+    # should work, but doesn't because tox.ini doesn't apply cleanly
+    # and fetchpatch with excludes doesn't seem to have any effect
+    ./fix-tests-no-tox-ini.patch
+  ];
+
+  propagatedBuildInputs = [
+    click
+    jinja2
+    markdown
+    mergedeep
+    pyyaml
+    pyyaml-env-tag
+    ghp-import
+    importlib-metadata
+    watchdog
+    packaging
+  ];
+
+  checkInputs = [
+    Babel
+    mock
+    pytestCheckHook
+  ];
+
+  pytestFlagsArray = [ "mkdocs/tests/*.py" ];
+
+  disabledTests = [
+    # Don't start a test server
+    "testing_server"
```

Comment:
```
_______________________ ERROR at setup of testing_server _______________________
file /build/source/mkdocs/tests/livereload_tests.py, line 31
  @contextlib.contextmanager
  def testing_server(root, builder=lambda: None, mount_path="/"):
E       fixture 'root' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

Please check where the fixture is supposed to come from.

Reply:
We can't use `pytestCheckHook` it seems, since test discovery assumes that any function with a `test_` prefix is a test, and any arguments therein a fixture.

Reply:
The upstream uses `python -m unittest discover`.